homeassistant/components/zha/core/channels/smartenergy.py

```diff
@@ -121,19 +190,33 @@ async def async_initialize_channel_specific(self, from_cache: bool) -> None:
         fmting = self.cluster.get(
             "demand_formatting", 0xF9
         )  # 1 digit to the right, 15 digits to the left
+        self._format_spec = self.get_formatting(fmting)
 
-        r_digits = int(fmting & 0x07)  # digits to the right of decimal point
-        l_digits = (fmting >> 3) & 0x0F  # digits to the left of decimal point
+        fmting = self.cluster.get(
+            "demand_formatting", 0xF9
```

Comment:
This is format for current_summ_delivered? Then  `"demand_formatting"` --> `"summa_formatting"`?
One digit to the right of the decimal point is probably adequate for metering of aggregate consumption in a house, but is perhaps a bit too coarse a default for plugs metering the energy consumption of individual devices? Could 2 decimal places be a reasonable default?

Reply:
No. Demand formatting depends on demand formatting attribute value and summa formatting depends on summa formatting value. Those could be different, could be same

Reply:
> 
> 
> No. Demand formatting depends on demand formatting attribute value and summa formatting depends on summa formatting value. Those could be different, could be same

@Adminiuga, sorry for not including more lines of context. But you seem to reference "demand_formatting" for both demand formatting _and_ for summa formatting, no?

Reply:
Both are calculated using the same algorithm, but stored in different attributes for later. Check the metering channel source, diff could be hard to see the whole picture.

Although, because currently the value is translated into W instead of KW, the formatting may be moot for some devices.

Reply:
I checked the source, and from what I can see the input to get_formatting method (for determining the summa format) is currently retrieved from the `demand_formatting` attribute (0x0304); not from the `summa_formatting` attribute (0x0303). Am I looking in the wrong place?

Reply:
Doh, you are right. Copy'n'paste. This was supposed to be summa formatting. Thank you for catching this.

Reply:
IMO should let HA to do KW to W conversion, instead of doing it in the channel